File: mobile/lib/core/common/features/domain/entities/question_entity.dart
import 'package:equatable/equatable.dart';
import 'answer_entity.dart';
import 'chapter_entity.dart';

class QuestionEntity extends Equatable {
  const QuestionEntity({
    required this.id,
    required this.index,
    required this.text,
    this.image,
    required this.explain,
    this.type,
    this.isImportant = false,
    required this.vehicle,
    this.chapter,
    this.answers = const [],
  });

  final int id;
  final int index;
  final String text;
  final String? image;
  final String? explain;
  final String? type;
  final bool isImportant;
  final String vehicle;
  final ChapterEntity? chapter;
  final List<AnswerEntity>? answers;

  @override
  List<Object?> get props => [
    id,
    index,
    text,
    image,
    explain,
    type,
    isImportant,
    vehicle,
    chapter,
    answers,
  ];
}

File: server/src/quizzes/entities/question.entity.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    OneToMany, ManyToMany,
} from 'typeorm';
import {Chapter} from './chapter.entity';
import {Answer} from './answer.entity';
import {Quiz} from "./quiz.entity";

@Entity('question')
export class Question {
    @PrimaryGeneratedColumn()
    id: number;

    @Column()
    index: number;

    @Column()
    text: string;

    @Column({nullable: true})
    image: string;

    @Column({nullable: true})
    explain: string;

    @Column({default: false})
    isImportant: boolean;

    @Column()
    vehicle: string;

    @ManyToOne(() => Chapter, (chapter) => chapter.questions)
    chapter: Chapter;

    @OneToMany(() => Answer, (answer) => answer.question)
    answers: Answer[];

    @ManyToMany(() => Quiz, (quiz) => quiz.questions)
    quizzes: Quiz[];
}
